python/lsst/meas/mosaic/updateExposure.py

```diff
@@ -190,8 +191,8 @@ def applyMosaicResultsCatalog(dataRef, catalog, addCorrection=True):
     # Now rotate them back to the LSST coord system
     if hscRun is None:
         if nQuarter%4 != 0:
-            catalog = mosaicUtils.rotatePixelCoordsBack(catalog, calexp_md.get("NAXIS1"),
-                                                        calexp_md.get("NAXIS2"), nQuarter)
+            catalog = mosaicUtils.rotatePixelCoordsBack(catalog, dimensions.getX(),
+                                                        dimensions.getY(), nQuarter)
```

Comment:
I realize this code violates linting all over the place, but can you do anything to make sure it won't complain about `dimensions` possibly not being defined here? If it's too expensive to always get `dimensions` even if it's not needed, then you could set it to None before the first `if nQuarter%4...`

Reply:
I ran a linter on this file and found one real error which I will ask you to fix while you're in here: TaskError is not defined. Please add it to the items imported form lsst.pipe.base at line 26.

Reply:
The function `getFluxFitParams` gets `wcsHeader` but does not use it. I suggest deleting the two lines that get it. I suspect this may be leftover

Reply:
The function `getWcs` reads metadata and then uses it to construct a WCS. Would it work to read the WCS directly?: 
```
    if hscRun is not None:
       wcsName = "wcs_hsc"
    else:
        wcsName = "wcs"
    return dataRef.get(wcsName, immediate=True)
```

Reply:
Unfortunately the `wcs` dataset is actually an `Exposure` dataset, so at best we could do dataRef.get("wcs").getWcs()` - but I *think* even that doesn't work because it's a zero-size `Exposure`.

We should make it actually a `Wcs` (or soon, `SkyWcs`) persisted directly with `FitsCatalogStorage` ([DM-11138](https://jira.lsstcorp.org/browse/DM-11138)).  I think that'll have to be backwards incompatible, however, and I haven't had the free energy to RFC that and push it through.